android/app/src/main/java/edu/berkeley/eecs/e_mission/SettingsActivity.java

```diff
@@ -402,5 +417,25 @@ public void showAllTrips(View view)
 		intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
 		startActivity(intent);
     }
+
+	public void onItemSelected(AdapterView<?> parent, View view,
+							   int pos, long id) {
+		// An item was selected. You can retrieve the selected item using
+		// parent.getItemAtPosition(pos)
+        Log.d(TAG, "Entered the onItemSelected function");
+        try {
+            String chosenBackend = String.valueOf(parent.getItemAtPosition(pos).toString());
+            Log.d(TAG, "Chosen Backend is " + chosenBackend);
+            ChosenBackend = chosenBackend;
```

Comment:
This is not a valid solution because the activity can be killed at any time by the system to reduce memory pressure. You need to store this somewhere. SharedPreferences is a reasonable solution.